tests/operators/test_read.py

```diff
@@ -240,3 +240,36 @@ def test_lfric_time_coord_fix_callback_no_time():
     cube = iris.cube.Cube([0, 0, 0], aux_coords_and_dims=[(length_coord, 0)])
     read._lfric_time_coord_fix_callback(cube, None, None)
     assert len(cube.coords("time")) == 0
+
+
+def test_pressurecoordfix_callback():
+    """Check that pressure_level is renamed to pressure if it exists."""
+    cube = iris.load_cube("tests/test_data/transect_test_umpl.nc")
+    cube.coord("pressure").rename("pressure_level")
+    read._fix_pressurecoord_name_callback(cube)
+    assert (
+        str(cube.coords)
+        == "<bound method Cube.coords of <iris 'Cube' of air_temperature / (K) (time: 2; pressure: 16; latitude: 6; longitude: 6)>>"
+    )
+
+
+def test_spatialcoordrename_callback():
+    """Check that spatial coord gets renamed if it is not grid_latitude."""
+    # This cube contains 'latitude' and 'longitude'
+    cube = iris.load_cube("tests/test_data/transect_test_umpl.nc")
+    read._fix_spatialcoord_name_callback(cube)
+    assert (
+        str(cube.coords)
+        == "<bound method Cube.coords of <iris 'Cube' of air_temperature / (K) (time: 2; pressure: 16; grid_latitude: 6; grid_longitude: 6)>>"
+    )
+
+
+def test_spatialcoordnotexist_callback():
+    """Check that spatial coord returns cube if cube does not contain spatial coordinates."""
+    cube = iris.load_cube("tests/test_data/transect_test_umpl.nc")
+    cube = cube[:, :, 0, 0]  # Remove spatial dimcoords
```

Comment:
I'm not convinced this actually removes the coords. As well as this you will need to call `cube.remove_coord("latitude")` and `cube.remove_coord("longitude")`.

Otherwise you will be left with two scalar coordinates, which are still valid latitudes and longitudes.